packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts

```diff
@@ -13,80 +19,44 @@ import.meta.hot = __vite__createHotContext(import.meta.url);
 if (import.meta.hot && !globalThis.__WAKU_HMR_CONFIGURED__) {
   globalThis.__WAKU_HMR_CONFIGURED__ = true;
   import.meta.hot.on('hot-import', (data) => import(/* @vite-ignore */ data));
-  const removeSpinner = () => {
-    const spinner = document.getElementById('waku-module-spinner');
-    spinner?.nextSibling?.remove();
-    spinner?.remove();
-  }
-  setTimeout(removeSpinner, 500);
   import.meta.hot.on('module-import', (data) => {
     // remove element with the same 'waku-module-id'
     let script = document.querySelector('script[waku-module-id="' + data.id + '"]');
+    let style = document.querySelector('style[waku-module-id="' + data.id + '"]');
     script?.remove();
     const code = data.code;
     script = document.createElement('script');
     script.type = 'module';
     script.text = code;
     script.setAttribute('waku-module-id', data.id);
     document.head.appendChild(script);
-    if (data.css) removeSpinner();
+    // avoid HMR flash by first applying the new and removing the old styles 
+    if (style) {
+      queueMicrotask(style.remove);
+    }
```

Comment:
I was wondering why you queued `style.remove` function?

Reply:
So we do not have a FOUC in HMR, if we don't do this, styles will be removed, for a small bit of time, there are no styles until the new script tag applies the new styles. By queuing, we first apply the new script styles and then after that we remove the old styles.